### PR TITLE
TYP: improve typing for DataFrame.to_string

### PR DESCRIPTION
**Pull request by topper-123**

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1043,6 +1041,56 @@ def _repr_html_(self) -> str | None:
         else:
             return None
 
+    @overload
+    def to_string(
+        self,
+        buf: None = None,
+        columns: Sequence[str] | None = None,
+        col_space: int | None = None,
```
Comment (phofl) [inline pandas/core/frame.py:1049]: Not directly related to your pr, but this type hint is wrong, can also be list[int] or dict

Comment (topper-123) [reply pandas/core/frame.py:1049]: Ok, I can look into that. Would like to get current version to pass first, then I’ll do this.

**Comment by jreback** [comment]
cc @simonjayhawkins @twoertwein if comments

File: pandas/core/frame.py [hunk 2]
```diff
@@ -1043,6 +1041,56 @@ def _repr_html_(self) -> str | None:
         else:
             return None
 
+    @overload
+    def to_string(
+        self,
+        buf: None = None,
```
Comment (simonjayhawkins) [inline pandas/core/frame.py:1047]: can leave out the defaults for the overloads. It's only the parameter types that are relevant in overloads.

```suggestion
        buf: None = ...,
```

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@topper-123 there was a parameter moved in #44331

need changes here to the overloads to prevent mypy errors on merge master.

```
pandas/core/frame.py:1094: error: Overloaded function implementation does not accept all possible arguments of signature 1  [misc]
pandas/core/frame.py:1094: error: Overloaded function implementation does not accept all possible arguments of signature 2  [misc]
```

**Comment by topper-123** [comment]
Updated.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @topper-123 LGTM

**Comment by topper-123** [comment]
Ping…

**Comment by simonjayhawkins** [comment]
Thanks @topper-123

**Comment by simonjayhawkins** [comment]
This is going to fail ci checks... looking now.

**Comment by simonjayhawkins** [comment]
FilePathOrBuffer was changed in #43951

**Pull request merged**